osu.Game/Rulesets/Scoring/ScoreStore.cs

```diff
@@ -49,8 +50,24 @@ public void Import(params string[] paths)
 
         public Score ReadReplayFile(string replayFilename)
         {
-            using (Stream s = storage.GetStream(Path.Combine(replay_folder, replayFilename)))
-                return new DatabasedLegacyScoreParser(rulesets, beatmaps).Parse(s);
+            Stream stream;
+            if (File.Exists(replayFilename))
+            {
+                // Handle replay with File since it is outside of storage
+                stream = File.OpenRead(replayFilename);
+            }
+            else if (storage.Exists(Path.Combine(replay_folder, replayFilename)))
```

Comment:
This fallback logic is really weird. Just remove it IMO.

Reply:
Remove the `storage.Exists` part?

Reply:
yes. it is never used.